app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -262,6 +262,14 @@ def add_middleware_deployment(ems_ref, hash)
           :binary_content        => hash[:file]["file"].read,
           :resource_path         => ems_ref.to_s
         }
+        unless hash[:file]['server_groups'].nil?
+          # in case of deploying into server group the resource path should point to the domain controller
+          deployment_data[:server_groups] = hash[:file]['server_groups']
+          server_group_path_hash = ::Hawkular::Inventory::CanonicalPath.parse(deployment_data[:resource_path]).to_h
+          server_group_path_hash[:resource_ids].slice!(1..-1)
```

Comment:
Ah okay thanks @Jiri-Kremser 

> Perhaps the future improvement would be to add the more readable methods

I think this would be a really good addition, would make it much more readable to us middleware novices :)